### PR TITLE
Improve readability of Operator guides

### PR DESCRIPTION
**Pull request by mabartos**
YAML attributes in the Operator guides are present in the bash command, which creates the YAML file. IMHO, it's not very well readable, and a plain YAML snippet fits better there with the help of syntax highlighting. 

**Example of provided changes**

Before:

After:

@vmuzikar @ahus1 WDYT?

### CODE REVIEW

File: docs/documentation/pom.xml [hunk 1]
```diff
@@ -145,7 +144,7 @@
                 <plugin>
                     <groupId>org.asciidoctor</groupId>
                     <artifactId>asciidoctor-maven-plugin</artifactId>
-                    <version>${version.plugin.asciidoctor}</version>
+                    <version>${asciidoctor.plugin.version}</version>
```
Comment (ahus1) [inline docs/documentation/pom.xml:147]: As the version is pinned via `pluginManagement` in the parent pom, the version line can (and should) be removed.

```suggestion
```

File: docs/guides/operator/basic-deployment.adoc [hunk 2]
```diff
@@ -114,11 +119,11 @@ kubectl create secret generic keycloak-db-secret \
   --from-literal=password=[your_database_password]
 ----
 
-You can customize several fields using the Keycloak CRD. However, for a basic deployment, you can enter the following example commands:
+You can customize several fields using the Keycloak CRD. For a basic deployment, you can stick to the following approach:
 
-[source,bash]
+Create YAML file *_'example-kc.yaml'_*:
```
Comment (ahus1) [inline docs/guides/operator/basic-deployment.adoc:124]:
```suggestion
Create YAML file `example-kc.yaml`:
```

File: docs/guides/operator/basic-deployment.adoc [hunk 3]
```diff
@@ -25,11 +25,11 @@ A database should be available and accessible from the cluster namespace where K
 For a list of supported databases, see <@links.server id="db"/>.
 The Keycloak Operator does not manage the database and you need to provision it yourself. Consider verifying your cloud provider offering or using a database operator such as https://access.crunchydata.com/documentation/postgres-operator/latest/[Crunchy].
 
-For development purposes, you can use an ephemeral PostgreSQL pod installation. To provision it, enter the following commands:
+For development purposes, you can use an ephemeral PostgreSQL pod installation. To provision it, follow the approach below:
 
-[source,bash]
+Create YAML file *_'example-postgres.yaml'_*:
```
Comment (ahus1) [inline docs/guides/operator/basic-deployment.adoc:30]:
```suggestion
Create YAML file `example-postgres.yaml`:
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
I approve the direction this is going. It helps the users to understand what is going on while it requires some more keystrokes on their side. Do we know any other documentation which uses the "pipe" strategy?

Once this PR is in, we would need to enable syntax highlighting on keycloak.org, as AFAIK it is not enabled there. Some additional parameters and CSS will be needed, and I'm happy to help with that. 

See below for some small review comments.

**Review by vmuzikar — COMMENTED** [review]
@mabartos Thanks for the PR. We used the pipe strategy back then to make it super simple for copy-pasting, but I agree a readability might be more important.

**Comment by mabartos** [comment]
> Do we know any other documentation which uses the "pipe" strategy?

@ahus AFAIK, this doc for Operator is the only one where it's used.

> Once this PR is in, we would need to enable syntax highlighting on keycloak.org, as AFAIK it is not enabled there. Some additional parameters and CSS will be needed, and I'm happy to help with that.

Great! Thanks for your help.

Btw. I'm afraid tests for documentation don't work properly. I was able to reproduce the failure even with the latest main branch, so it shouldn't be related to this PR. However, our CI for documentation/guides doesn't catch that; I'll analyze it.

File: docs/guides/operator/realm-import.adoc [hunk 4]
```diff
@@ -48,9 +48,9 @@ The recommended way to obtain a `RealmRepresentation` is by leveraging the expor
 
 Use `kubectl` to create the CR in the correct cluster namespace:
 
-[source,bash]
+Create YAML file `example-realm-import.yaml`:
+[source,yaml]
```
Comment (ahus1) [inline docs/guides/operator/realm-import.adoc:52]:
```suggestion
Create YAML file `example-realm-import.yaml`:

[source,yaml]
```

File: docs/guides/operator/basic-deployment.adoc [hunk 5]
```diff
@@ -169,9 +179,9 @@ CONDITION: RollingUpdate
 The Keycloak deployment is exposed through a basic Ingress and is accessible through the provided hostname.
 If the default ingress does not fit your use case, disable it by setting `ingress` spec with `enabled` property to `false` value:
 
-[source,bash]
+Edit YAML file `example-kc.yaml`:
+[source, yaml]
```
Comment (ahus1) [inline docs/guides/operator/basic-deployment.adoc:183]:
```suggestion
Edit YAML file `example-kc.yaml`:

[source,yaml]
```

File: docs/guides/operator/realm-import.adoc [hunk 6]
```diff
@@ -62,7 +62,11 @@ spec:
     realm: example-realm
     displayName: ExampleRealm
     enabled: true
-EOF
+----
+
+Apply the changes:
+[source,bash]
+----
```
Comment (ahus1) [inline docs/guides/operator/realm-import.adoc:69]:
```suggestion
Apply the changes:

[source,bash]
----
```

File: pom.xml [hunk 7]
```diff
@@ -35,6 +35,8 @@
     <packaging>pom</packaging>
 
     <properties>
+        <asciidoctor.plugin.version>2.2.3</asciidoctor.plugin.version>
```
Comment (ahus1) [inline pom.xml:38]: This is a major upgrade, which upgrades to an Asciidoctor version which is more ahead than the downstream version of Asciidoctor, therefore please keep the old version here. 

I'm aligning the versions a bit with https://github.com/keycloak/keycloak/pull/20266, but not so much. Again, please revert the changes on `pom.xml` for this PR. It needs a bit more discussion.

**Review by ahus1 — CHANGES_REQUESTED** [review]
I added some blank lines to align with common AsciiDoc practices. The major thing: Please don't upgrade the major version of the plugin, as it might break some things upstream/downstream or on Keycloak Web. Thanks!

File: pom.xml [hunk 8]
```diff
@@ -35,6 +35,8 @@
     <packaging>pom</packaging>
 
     <properties>
+        <asciidoctor.plugin.version>1.5.8</asciidoctor.plugin.version>
```
Comment (mabartos) [inline pom.xml:38]: @ahus1 We can at least use the 1.5.8, right?

Comment (ahus1) [reply pom.xml:38]: @mabartos - yes, 1.5.8 is fine.

**Review by ahus1 — APPROVED** [review]
Thank you, looks great.

**Review by vmuzikar — APPROVED** [review]
